src/jdk.internal.le/share/classes/jdk/internal/console/JdkConsoleImpl.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2022, 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package jdk.internal.console;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.Locale;
+import jdk.internal.console.SimpleConsoleReader.CleanableBuffer;
+
+import jdk.internal.io.BaseJdkConsoleImpl;
+
+/**
+ * JdkConsole implementation based on the platform's TTY, with basic keyboard navigation.
+ */
+public final class JdkConsoleImpl extends BaseJdkConsoleImpl {
```

Comment:
We should choose a different simple name for this class, especially when we are importing classes from `jdk.internal.io` package.